### PR TITLE
Added routing and shell component for the metering points page

### PR DESCRIPTION
**Pull request by Klemmensen**
### Metering points
- Add page

### Dashboard
- Add message to the Dashboard

### Landing page
- Update message

### Shared
- Add a title store
- Add Inline Message component

### Tests
- Update end-to-end tests

## References
- https://github.com/Energinet-DataHub/energy-origin/issues/412

Feature environment: https://eo-u-meteringpoints.westeurope.cloudapp.azure.com/

### CODE REVIEW

File: libs/eo/dashboard/shell/.browserslistrc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# This file is used by the build system to adjust CSS and JS output to support the specified browsers below.
```
Comment (LayZeeDK) [inline libs/eo/dashboard/shell/.browserslistrc:1]: nitpick: This configuration can be deleted.

File: libs/eo/dashboard/shell/src/lib/eo-dashboard-shell.component.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        @include watt.typography-watt-text-m; // This overrides the styles applied from Angular Material on p tags
+      }
+      .${selector}__eo-inline-message {
+        margin-bottom: var(--watt-space-l);
+      }
+    `,
+  ],
+  template: `
+    <eo-inline-message
+      icon="/assets/images/icons/primary-info-icon.svg"
```
Comment (LayZeeDK) [inline libs/eo/dashboard/shell/src/lib/eo-dashboard-shell.component.ts:48]: suggestion: Use the Watt Icon componetn and add this SVG icon to the `MatIconRegistry`.

File: libs/eo/dashboard/shell/src/lib/eo-dashboard-shell.component.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      >.
+    </p>
+  `,
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EoDashboardShellComponent {
+  constructor(private eoTitleService: EoTitle) {
+    this.eoTitleService.setTitle('Dashboard');
+  }
```
Comment (LayZeeDK) [inline libs/eo/dashboard/shell/src/lib/eo-dashboard-shell.component.ts:74]: issue: Let's see if we can improve how the page title is managed, for example by subscribing to Router events.

File: libs/eo/landing-page/shell/src/lib/eo-landing-page.store.ts [hunk 4]
```diff
@@ -28,7 +27,7 @@ interface EoLandingPageState {}
 
 @Injectable()
 export class EoLandingPageStore extends ComponentStore<EoLandingPageState> {
-  #absoluteReturnUrl = this.urlGenerator.fromCommands([ettDashboardRoutePath]);
+  #absoluteReturnUrl = this.urlGenerator.fromCommands(['dashboard']);
```
Comment (LayZeeDK) [inline libs/eo/landing-page/shell/src/lib/eo-landing-page.store.ts:30]: issue: This route path should be located in a routing library.

File: libs/eo/landing-page/shell/src/lib/layout/eo-landing-page-notification.component.ts [hunk 5]
```diff
@@ -33,50 +35,24 @@ const selector = 'eo-landing-page-notification';
...
-          NemID/MitID is intended to form part of one of the next releases.
-        </p>
-      </div>
-    </div>
+    <eo-inline-message
+      icon="/assets/images/icons/primary-info-icon.svg"
+      class="${selector}__eo-inline-message"
+    >
+      <p>
+        The Energy Origin Platform is <b>under development</b> and new
```
Comment (LayZeeDK) [inline libs/eo/landing-page/shell/src/lib/layout/eo-landing-page-notification.component.ts:50]: issue: Don't use the `<b>` tag, it was deprecated 14 years ago 🙂 Look for uses in other pages.
```suggestion
        The Energy Origin Platform is <strong>under development</strong> and new
```

File: libs/eo/metering-points/shell/src/lib/eo-metering-points-shell.component.ts [hunk 6]
```diff
@@ -35,23 +35,24 @@ const selector = 'ett-dashboard-shell';
...
   template: `
-    <h2>Dashboard</h2>
-    <p class="${selector}__p">
-      More functionality will be released on an ongoing basis. If you want to
-      influence the new functionality, please send an email to
-      <a href="mailto:xkeka@energinet.dk">xkeka@energinet.dk</a>.
-    </p>
+    <p class="${selector}__p">You do not have any metering points.</p>
   `,
+  encapsulation: ViewEncapsulation.None,
```
Comment (LayZeeDK) [inline libs/eo/metering-points/shell/src/lib/eo-metering-points-shell.component.ts:45]: suggestion: We can stop using `ViewEncasulation.None` and instead rely on emulated view encapsulation (the default setting). Same for other components introduced by this pull request.

File: libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  NgModule,
+  ViewEncapsulation,
+} from '@angular/core';
+
+import { CommonModule } from '@angular/common';
+
+export type inlineMessageType =
```
Comment (LayZeeDK) [inline libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts:28]: issue: We use PascalCase to name types.

File: libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+  template: `
+    <div
+      class="${selector}__container"
+      [ngClass]="{
+        info: type === 'info',
+        success: type === 'success',
+        danger: type === 'danger',
+        warning: type === 'warning',
+        default: type === 'default'
+      }"
```
Comment (LayZeeDK) [inline libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts:89]: suggestion: How about a simple template expression?
```suggestion
      class="${selector}__container {{ type }}"
```

File: libs/ett/core/shell/src/lib/ett-core-shell.module.ts [hunk 9]
```diff
@@ -45,10 +45,17 @@ const routes: Routes = [
     canActivateChild: [EttAuthenticationGuard],
     children: [
       {
-        path: ettDashboardRoutePath,
+        path: 'dashboard',
```
Comment (LayZeeDK) [inline libs/ett/core/shell/src/lib/ett-core-shell.module.ts:48]: issue: This route path should be located in a routing library.

File: libs/ett/core/shell/src/lib/ett-primary-navigation.component.ts [hunk 10]
```diff
@@ -66,6 +67,13 @@ const selector = 'ett-primary-navigation';
       <a mat-list-item routerLink="/dashboard" routerLinkActive="is-active">
         Dashboard
       </a>
+      <a
+        mat-list-item
+        routerLink="/metering-points"
+        routerLinkActive="is-active"
+      >
+        Metering points
+      </a>
```
Comment (LayZeeDK) [inline libs/ett/core/shell/src/lib/ett-primary-navigation.component.ts:76]: suggestion: Consider using the Watt Nav List component.

File: libs/ett/shared/util-browser/src/lib/eo-title.service.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { Title } from '@angular/platform-browser';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EoTitle {
+  private titleSubject = new BehaviorSubject<string>('');
+  title$!: Observable<string>;
```
Comment (LayZeeDK) [inline libs/ett/shared/util-browser/src/lib/eo-title.service.ts:26]: issue: We don't use non-null assertions. We should have/find a lint rule for this 🤔

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

File: libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+      </div>
+      <div class="${selector}__content">
+        <ng-content></ng-content>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class EoInlineMessageComponent {
```
Comment (LayZeeDK) [inline libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts:107]: issue: A reusable component should be covered with component tests.

File: libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts (hunk 8 above)
Comment (Klemmensen) [reply libs/eo/shared/atomic-design/ui-atoms/src/lib/eo-inline-message/eo-inline-message.component.ts:89]: Much simpler to read, agree

File: libs/ett/shared/util-browser/src/lib/eo-title.store.ts [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+import { filter, map, Observable, tap } from 'rxjs';
+
+interface EoTitleState {
+  readonly routeTitle: string | null;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EoTitleStore extends ComponentStore<EoTitleState> {
```
Comment (LayZeeDK) [inline libs/ett/shared/util-browser/src/lib/eo-title.store.ts:30]: issue: This class must be covered with unit tests.

**Review by LayZeeDK — APPROVED** [review]
🚀

**Pull request merged**